docs/sql-migration-guide.md

```diff
@@ -43,7 +43,7 @@ license: |
 
   - The `ADD JAR` command previously returned a result set with the single value 0. It now returns an empty result set.
 
-  - In Spark version 2.4 and earlier, users can create map values with map type key via built-in function like `CreateMap`, `MapFromArrays`, etc. Since Spark 3.0, it's not allowed to create map values with map type key with these built-in functions. Users can still read map values with map type key from data source or Java/Scala collections, though they are not very useful.
+  - In Spark version 2.4 and earlier, users can create map values with map type key via built-in function like `CreateMap`, `MapFromArrays`, etc. Since Spark 3.0, it's not allowed to create map values with map type key with these built-in functions, users can use array<struct<key, value>> as map keys instead. Users can still read map values with map type key from data source or Java/Scala collections, though they are not very useful.
```

Comment:
> ... built-in functions, users can use array<struct<key, value>> as map keys instead. Users can still read map values with map type key from data source or Java/Scala collections, though they are not very useful.

I would write as below:

... built-in functions. Users can use `array<struct<key, value>>` as map keys instead as a workaround. In addition, users can still read map values with map type key from data source or Java/Scala collections, though they are not very useful.

Reply:
Maybe .. "though they are not very useful." -> "though it is discouraged." (?).